LEAF_Request_Portal/templates/reports/LEAF_mass_action.tpl

```diff
@@ -0,0 +1,452 @@
+    }
+
+    //when filtering by resolved (either = or !=), 
+    //form/query also makes sure that: requests ARE submitted and AREN'T cancelled
+    //so, here we make sure that these are already the case
+    if(getResolved === 'true' && getSubmitted === 'true' && getCancelled === 'false')
+    {
+        requestQuery.terms.push({"id":"stepID","operator":"=","match":"resolved"});
+    }
+    else if(getResolved === 'false' && getSubmitted === 'true' && getCancelled === 'false')
+    {
+        requestQuery.terms.push({"id":"stepID","operator":"!=","match":"resolved"});
+    }
+    return requestQuery;
+}
+
+/**
+ * Looks up requests based on filter/searchbar and builds table with the results
+ *
+ * @param {Object}  [queryObj]  Object to pass to form/query
+ */
+function listRequests(queryObj)
+{
+    $('#searchResults').hide();
+    $('#errorMessage').hide();
+    $('table#requests tr.requestRow').remove();
+    $('#iconBusy').show();
+
+    $.ajax({
+        type: 'GET',
+        url: './api/?a=form/query',
+        data: {q: JSON.stringify(queryObj),
+                CSRFToken: '<!--{$CSRFToken}-->'},
+        cache: false
+    }).done(function(data) {
+        if(Object.keys(data).length)
+        {
+            $.each(data, function( index, value ) {
+                requestsRow = '<tr class="requestRow">';
+                requestsRow += '<td><a href="index.php?a=printview&amp;recordID='+value.recordID+'">'+value.recordID+'</a></td>';
+                requestsRow += '<td>'+value.categoryNames[0]+'</td>';
+                requestsRow += '<td>'+(value.service == null ? '' : value.service)+'</td>';
+                requestsRow += '<td>'+value.title+'</td>';
+                requestsRow += '<td><input type="checkbox" name="massActionRequest" class="massActionRequest" value="'+value.recordID+'"></td>';
+                requestsRow += '</tr>';
+                $('table#requests').append(requestsRow);
+            });
+            $('#searchResults').show();
+        }
+        else
+        {
+            $('#errorMessage').html('No Results');
+            $('#errorMessage').show();
+        }
+    }).fail(function (jqXHR, error, errorThrown) {
+        console.log(jqXHR);
+        console.log(error);
+        console.log(errorThrown);
+    }).always(function (){
+        $('#iconBusy').hide();
+    });
+}
+
+/**
+ * Executes the selected action on each request selected in the table
+ */
+function executeMassAction()
+{
+    var selectedRequests = $('input.massActionRequest:checked');
+    processedRequests = 0;
+    totalActions = selectedRequests.length;
+    successfulActionRecordIDs = [];
+    failedActionRecordIDs = [];
+    
+    if(totalActions)
+    {
+        $('button.takeAction').attr("disabled", "disabled");
+    }
+    $.each(selectedRequests, function(key, item) {
+        var ajaxPath = '';
+        var ajaxData = {};
+        var recordID = $(item).val();
+        switch(actionValue) {    
+            case 'submit':
```

Comment:
When a request is submitted with an error, the UI says it was a success.  Specifically, a request tied to an incorrectly configured workflow (like a workflow that that has no finish point) says the submission was successful, even though the network log says there was an error.

Reply:
This will be addressed next sprint.